### PR TITLE
add TryLock method for non-blocking clients

### PR DESCRIPTION
**Pull request by natebrennand**
Add a `TryLock` method to mirror the `TryRLock` method that already exists.

This method is for use in an application that would like to control their own retry
behavior instead of a blocking request that loops.

### CODE REVIEW

File: rw_mutex.go [hunk 1]
```diff
@@ -39,6 +39,43 @@ func NewRWMutex(collection *mgo.Collection, lockID, clientID string) *RWMutex {
...
+		"$set": bson.M{
+			"writer": m.clientID,
+		},
+	})
+	if err == nil {
+		return nil // we got the lock!
+	} else if err != mgo.ErrNotFound {
+		// This only works if the mgo.Session object has Safe mode enabled.
+		// Safe is the default but something for which we should maintain
+		// external documentation
```
Comment (prime-time) [inline rw_mutex.go:57]: Thanks for the helpful comment

Comment (natebrennand) [reply rw_mutex.go:57]: That comment was just copy/pasted... I actually don't know what it refers to... :hide:

**Review by prime-time — APPROVED** [review]
🔒  looks good to me  🗝

File: rw_mutex.go (hunk 1 above)
Comment (prime-time) [reply rw_mutex.go:57]: ©️ 🍝

Comment (natebrennand) [reply rw_mutex.go:57]: Looks like Safe is on by default - https://github.com/go-mgo/mgo/blob/v2/session.go#L534
So a consumer would have to explicitly have overwritten this setting.

**Pull request merged**